src/util/bip32.rs

```diff
@@ -126,14 +126,14 @@ impl ChildNumber {
     /// Returns `true` if the child number is a [`Normal`] value.
     ///
     /// [`Normal`]: #variant.Normal
-    pub fn is_normal(self) -> bool {
+    pub fn is_normal(&self) -> bool {
```

Comment:
nit: There's no need to do this. It's probably microscopically more efficient to leave it `self`.

Reply:
This is not about efficiency, this is required by compiler, otherwise we get

```console
error[E0631]: type mismatch in function arguments
   --> src/util/psbt/map/global.rs:182:71
    |
182 |                         (fp, deriv.len(), deriv.into_iter().rposition(ChildNumber::is_normal))
    |                                                                       ^^^^^^^^^^^^^^^^^^^^^^ expected signature of `fn(&ChildNumber) -> _`
    | 
   ::: src/util/bip32.rs:129:5
    |
129 |     pub fn is_normal(self) -> bool {
    |     ------------------------------ found signature of `fn(ChildNumber) -> _`

```

Reply:
you can just add `.cloned()` on the iterator before `.rposition`. Or `.copied()` in Rust 1.36. But what you did here is fine.